src/main/java/com/bunq/sdk/json/PaginationAdapter.java

```diff
@@ -0,0 +1,99 @@
+package com.bunq.sdk.json;
+
+import com.bunq.sdk.http.Pagination;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import java.lang.reflect.Type;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.charset.Charset;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.utils.URLEncodedUtils;
+
+/**
+ * Custom (de)serialization of SessionServer required due to the specific way we handle pagination.
+ */
+public class PaginationAdapter implements JsonDeserializer<Pagination> {
+
+  /**
+   * Field constants.
+   */
+  private static final String FIELD_OLDER_URL = "older_url";
+  private static final String FIELD_NEWER_URL = "newer_url";
+  private static final String FIELD_FUTURE_URL = "future_url";
+
+  @Override
+  public Pagination deserialize(JsonElement json, Type typeOfT,
+      JsonDeserializationContext context) throws JsonParseException {
+    try {
+      JsonObject responseJson = json.getAsJsonObject();
```

Comment:
This `try` block is kinds big, wouldnt it be a beter idea to only put the line that will throw the exception in the `try` block and put the rest in the `else` block?

Or are you expecting the exception on all these lines ?

Reply:
What kind of "else"?

Reply:
Anyway, I'll just extract a method from it and it'll become smaller :)

Reply:
```python
try:
    b
except:
    do stuff if exception caught 
else:
    do stuff if no exception
```

This way it's clear what is throwing the exception. That is what I meant with the `else's not sure if java has such thing though 🙈 just took a guess here 😁